### PR TITLE
feat: add java format plugin

### PR DESCRIPTION
**Pull request by caicancai**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?
add java format plugin

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Comment by caicancai** [comment]
If this PR is not suitable, I can close it. I just added a java code format plug-in

File: java/pom.xml [hunk 1]
```diff
@@ -84,6 +84,7 @@
         <datastax.version>4.15.0</datastax.version>
         <flink.version>1.18.0</flink.version>
         <testcontainers.version>1.17.6</testcontainers.version>
+        <maven-spotless-plugin.version>2.41.1</maven-spotless-plugin.version>
```
Comment (caicancai) [inline java/pom.xml:87]: this

**Comment by tabVersion** [comment]
Thanks for your contribution, I've involved maintainers to review.

**Comment by wenym1** [comment]
We already have our format plugin, which is exactly the one added in this PR. Currently we use the version `2.27.1`. If there is no special reason, we prefer retaining the current version. Our current plugin is set in https://github.com/risingwavelabs/risingwave/blob/main/java/pom.xml#L387-L410. For style setting you can refer to https://github.com/risingwavelabs/risingwave/tree/main/java/tools/maven

You can run `mvn spotless:apply` in the `java` folder to format the code.

**Comment by caicancai** [comment]
> We already have our format plugin, which is exactly the one added in this PR. Currently we use the version `2.27.1`. If there is no special reason, we prefer retaining the current version. Our current plugin is set in https://github.com/risingwavelabs/risingwave/blob/main/java/pom.xml#L387-L410. For style setting you can refer to https://github.com/risingwavelabs/risingwave/tree/main/java/tools/maven
> 
> You can run `mvn spotless:apply` in the `java` folder to format the code.

thank you，closing

**Pull request closed without merging**